openedx/features/enterprise_support/tests/test_api.py

```diff
@@ -266,28 +272,55 @@ def mock_get_enterprise_customer(**kwargs):
         # the third-party auth pipeline has no `provider_id`.
         mock_registry.get_from_pipeline.return_value.provider_id = None
         self.mock_get_enterprise_customer('real-ent-uuid', {'real': 'enterprisecustomer'}, 200)
-        enterprise_customer = enterprise_customer_for_request(
-            mock.MagicMock(GET={'enterprise_customer': 'real-ent-uuid'}, user=self.user)
+        mock_request = mock.MagicMock(
+            GET={'enterprise_customer': 'real-ent-uuid'},
+            COOKIES={},
+            session={},
+            user=self.user
         )
+        enterprise_customer = enterprise_customer_for_request(mock_request)
         self.assertEqual(enterprise_customer, {'real': 'enterprisecustomer'})
 
         # Verify that the method `enterprise_customer_for_request` returns
         # expected enterprise customer against the requesting user even if
         # the third-party auth pipeline has no `provider_id` but there is
         # enterprise customer UUID in the cookie.
-        enterprise_customer = enterprise_customer_for_request(
-            mock.MagicMock(GET={}, COOKIES={settings.ENTERPRISE_CUSTOMER_COOKIE_NAME: 'real-ent-uuid'}, user=self.user)
+        mock_request = mock.MagicMock(
+            GET={},
+            COOKIES={settings.ENTERPRISE_CUSTOMER_COOKIE_NAME: 'real-ent-uuid'},
+            session={},
+            user=self.user
         )
+        enterprise_customer = enterprise_customer_for_request(mock_request)
+        self.assertEqual(enterprise_customer, {'real': 'enterprisecustomer'})
+
+        # Verify that the method `enterprise_customer_for_request` returns
```

Comment:
Not on you, but this test is a lot

Reply:
Oh, thank you, I meant to DDT-ify this.

Reply:
On second-thought, that's a little too much to try to unpack right now, this test really depends on the order of operations, somehow.

Reply:
Ok that's fine